models/phylogeny/tree_inference_pruning.tppl

```diff
@@ -117,23 +126,22 @@ function cluster(q: Matrix[Real], trees: MsgTree[], maxAge: Real, seq_len: Int)
     let log_likes = sapply(node_msg, get_log_likes);
     logWeight (seqSumReal(log_likes));
 
-    // Remove weights of previous internal nodes
-    if (left_child is Node) {
-        let log_likes_left = sapply(left_child.msg, get_log_likes);
-        logWeight -(seqSumReal(log_likes_left));
-    }
-    if (right_child is Node) {
-        let log_likes_right = sapply(right_child.msg, get_log_likes);
-        logWeight -(seqSumReal(log_likes_right));
-    }
+    // Remove weights of previous internal children
+    let log_likes_left = sapply(left_child.msg, get_log_likes);
+    logWeight -(seqSumReal(log_likes_left));
+    let log_likes_right = sapply(right_child.msg, get_log_likes);
+    logWeight -(seqSumReal(log_likes_right));
+
+    // Manual resample
+    resample;
 
     // Combine picked pair of trees into a new node
     let parent = Node{age=age, msg=node_msg, left=left_child, right=right_child};
 
     // Compute new_trees list
-    let min = pairs[2];
-    let max = pairs[1];
-    let new_trees = paste0([slice(trees, 1, min), slice(trees, min + 1, max), slice(trees, max + 1, n + 1), [parent]]);
+    let min = mini(pairs[2],pairs[1]);
+    let max = maxi(pairs[2],pairs[1]);
```

Comment:
I don't think `mini` and `maxi` are exposed, so you probably need to add them. Should be just adding
```
let mathMini = mini
```
to `src/lib/standard.mc` and
```
function minInt(a : Int, b : Int) => Int {
  return mathMini(a)(b);
}
```
to `src/lib/standard.tppl`, and corresponding for `maxi`. Note the rename to `minInt`, to follow naming for `seqSumInt` and the like.